uint32 posX
uint32 posY
